src/pages/docs/guides/ai-transport/openai-sdk.mdx

```diff
@@ -0,0 +1,670 @@
+---
+title: "Guide: Token streaming using Ably and the OpenAI SDK"
+meta_description: "Stream AI model responses from the OpenAI SDK over Ably with real-time token delivery."
+meta_keywords: "AI, token streaming, OpenAI, Responses API, AI transport, Ably, realtime"
+---
+
+This guide demonstrates how to stream token-by-token responses from OpenAI's Responses API over Ably's realtime messaging platform. This pattern enables you to broadcast AI model responses to multiple clients simultaneously, with clear message boundaries for handling response lifecycle.
+
+By using Ably to distribute tokens from the OpenAI SDK, you can scale to thousands of concurrent subscribers, decouple your AI inference from client connections, and maintain reliable message delivery with ordering guarantees. This guide implements the [message-per-token with explicit start/stop events](/docs/ai-transport/features/token-streaming/message-per-token#explicit-events) pattern, which provides clear lifecycle management for each AI response.
+
+<Aside data-type="note">
+This guide focuses on OpenAI's Responses API. For general concepts about token streaming patterns, see the [message-per-token](/docs/ai-transport/features/token-streaming/message-per-token) guide.
+</Aside>
+
+## Prerequisites <a id="prerequisites"/>
+
+To follow this guide, you'll need:
+
+**Software requirements:**
+- Node.js 20 or higher
+
+**Account requirements:**
+- An OpenAI API key
+- An Ably API key
+
+**Useful links:**
+- [OpenAPI developer quickstart](https://platform.openai.com/docs/quickstart)
+- [Ably JavaScript SDK getting started](/docs/getting-started/javascript)
+
+## Setting up your environment <a id="setup"/>
```

Comment:
This section is usually under the prerequisites. For example here: https://ably.com/docs/chat/getting-started/javascript (I know we're not writing a getting started guide but it follows similar suit.